src/IR/IR.hs

```diff
@@ -252,3 +255,97 @@ wellFormed (Prim p es _) = wfPrim p es && all wellFormed es
   wfPrim (PrimOp PrimLt    ) [_, _]    = True
   wfPrim (PrimOp PrimLe    ) [_, _]    = True
   wfPrim _                   _         = False
+
+
+{-
+how to use:
+1) stack ghci
+2) import IR.IR
+3) samples:
+show (LitIntegral 3)
+show New
+Questions:
+why does `show (Lit "a" Integer)` not work? -- b/c types not defined, just use () as a type
+why does `show (Var (VarId (Identifier "foo")) LitIntegral)` not work?
+
+show (Var (VarId (Identifier "foo")) ()))
+
+-}
+instance (Pretty t) => Show (Expr t) where
```

Comment:
The `Show` instance should be derived, and the `Pretty` instance should be kept separate.

In my mind, `Show` should really dump the Haskell representation of a data structure (and it should be the inverse of `Read`).

Meanwhile, `pretty` should be the inverse of `parse`, modulo compiler transformations